src/gui_common/SegmentedBar.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A ProgressBar that is split up into IconProgressBars, data is stored in a dictionary
+/// </summary>
+public class SegmentedBar : Control
+{
+    public PackedScene IconProgressBarScene = GD.Load<PackedScene>("res://src/gui_common/IconProgressBar.tscn");
+    public string Type;
+    public float MaxValue;
+
+    public Vector2 Size
+    {
+        get
+        {
+            return new Vector2(RectSize.x, RectSize.y);
+        }
+        set
+        {
+            RectSize = new Vector2(value[0], value[1]);
+            RectMinSize = RectSize;
+        }
+    }
+
+    public void UpdateAndMoveBars(Dictionary<string, float> data)
+    {
+        RemoveUnusedBars(this, data);
+
+        int location = 0;
+        foreach (var dataPair in data)
+        {
+            CreateAndUpdateBar(dataPair, this, location);
+            location++;
+        }
+
+        foreach (var dataPair in data)
+        {
+            if (HasNode(dataPair.Key))
+                UpdateDisabledBars(dataPair, this);
+        }
+
+        foreach (var dataPair in data)
+        {
+            if (HasNode(dataPair.Key))
+                MoveBars(GetNode<IconProgressBar>(dataPair.Key));
+        }
+    }
+
+    private void RemoveUnusedBars(SegmentedBar parent, Dictionary<string, float> data)
+    {
+        List<IconProgressBar> unusedBars = new List<IconProgressBar>();
+        foreach (IconProgressBar progressBar in parent.GetChildren())
+        {
+            bool match = false;
+            foreach (var dataPair in data)
+            {
+                if (progressBar.Name == dataPair.Key)
+                    match = true;
+            }
+
+            if (!match)
+                unusedBars.Add(progressBar);
+        }
+
+        foreach (IconProgressBar unusedBar in unusedBars)
+            unusedBar.Free();
+    }
+
+    private void CreateAndUpdateBar(KeyValuePair<string, float> dataPair, SegmentedBar parent, int location = -1)
+    {
+        if (parent.HasNode(dataPair.Key))
+        {
+            IconProgressBar progressBar = (IconProgressBar)parent.GetNode(dataPair.Key);
+            if (location >= 0)
+            {
+                progressBar.SetBarLocation(location);
+                progressBar.SetBarActualLocation(location);
+            }
+
+            if (progressBar.Disabled)
+                return;
+            progressBar.SetBarLeftShift(
```

Comment:
I'm still of the opinion that messing with the margin, instead of just using a hbox with certain width children, is the wrong approach.